src/compiler/compile/nodes/Element.ts

```diff
@@ -395,7 +395,7 @@ export default class Element extends Node {
 					component.slot_outlets.add(name);
 				}
 
-				if (!(parent.type === 'InlineComponent' || (parent.type === 'Element' && /-/.test(parent.name)))) {
+				if (!(parent.type === 'InlineComponent' || within_custom_element(parent))) {
```

Comment:
we can actually allow it if it is within custom element, since we are going to create them just like a normal element, browser will figure out itself

Reply:
this relates to https://github.com/sveltejs/svelte/issues/1689#issuecomment-569422047

Reply:
Should the error message be adjusted then? Something like 'must be a direct child of a component or a descendant of a custom element'?

Reply:
yup! updated the error message